packages/api/src/util/animal.js

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import knex from './knex.js';
+
+/**
+ * Assigns internal identifiers to records based on the provided record IDs and kind.
```

Comment:
Maybe nit, but should this be called `getInternalIdentifiers` (and the JSDoc comment revised) as identifiers are not being assigned to records at all -- just generated when the view is called up?

Reply:
Oh sorry, I modified this function and forgot to update the JSDoc.

Identifiers are assigned on line 31: `record.internal_identifier = internalIdentifier`, so after this function is run, the records that were passed will have identifiers.
Initial records:
```
[{id: 1, name: "name"}, {id: 2, name: "name2"}]
```
Updated records:
```
[{id: 1, name: "name", internal_identifier: 1}, {id: 2, name: "name2", internal_identifier: 2}]
```

Reply:
Oh okay, I think we just had different definitions of `record` in mind. I thought a record was specifically a database term (an entry in a table), so a record update makes me think that a database update operation was performed.

But it sounds like here, a record is just a data object (or in context, the result object about to be sent to the client). Which I should have noticed, since it is the exact variable name you are using! 🙂 If that's clear to everyone else, that's fine with me too.